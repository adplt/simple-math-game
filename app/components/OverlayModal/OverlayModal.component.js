import React from 'react';
import PropTypes from 'prop-types';
import {View, Text} from 'react-native';
import Touchable from '../Touchable/Touchable.component';
import styles from './OverlayModal.component.style';
import Modal from 'react-native-modal';

class OverlayModal extends React.Component {

  static propTypes = {
    showModal: PropTypes.bool,
    closeModal: PropTypes.func,
    data: PropTypes.object,
  }

  state = {
    visible: false,
  }

  componentWillReceiveProps (newProps) {
    const {showModal} = newProps;
    this.setState({visible: showModal});
  }

  hideModal = () => {
    const {closeModal} = this.props;
    closeModal();
  }

  render () {
    const {closeModal, data} = this.props;
    const {visible} = this.state;
    return (
      <Modal
        isVisible={visible}>
        <View style={styles.container}>
          <View>
            <Text style={styles.title}>{'Time\'s Up'}</Text>
          </View>
          <View style={styles.content}>
            <View style={styles.row}>
              <Text style={styles.halfWidthRight}>{'Time: '}</Text>
              <Text style={styles.halfWidthLeft}>{'60'}</Text>
            </View>
            <View style={styles.row}>
              <Text style={styles.halfWidthRight}>{'Score: '}</Text>
              <Text style={styles.halfWidthLeft}>{data.score}</Text>
            </View>
            <View style={styles.row}>
              <Text style={styles.halfWidthRight}>{'High Score: '}</Text>
              <Text style={styles.halfWidthLeft}>{data.highScore}</Text>
            </View>
          </View>
          <Touchable onPress={closeModal}>
            <Text style={styles.close}>{'Close'}</Text>
          </Touchable>
        </View>
      </Modal>
    );
  }
}

export default OverlayModal;
