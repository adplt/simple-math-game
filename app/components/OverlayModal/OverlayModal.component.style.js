const halfWidth = {
  margin: 10,
  flex: 1,
};

export default {
  container: {
    flex: 0,
    backgroundColor: '#FFF',
    borderRadius: 30,
  },
  title: {
    fontSize: 40,
    textAlign: 'center',
    margin: 10,
    color: 'black',
    paddingVertical: 20
  },
  close: {
    fontSize: 30,
    textAlign: 'center',
    margin: 20
  },
  row: {
    flexDirection: 'row'
  },
  halfWidthLeft: [halfWidth, {textAlign: 'left'}],
  halfWidthRight: [halfWidth, {textAlign: 'right'}],
  content: {
    margin: 20
  }
};
