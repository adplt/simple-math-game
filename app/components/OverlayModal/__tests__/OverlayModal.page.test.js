import React from 'react';
import renderer from 'react-test-renderer';
import OverlayModal from '../OverlayModal.component';

describe('OverlayModal component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<OverlayModal />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
