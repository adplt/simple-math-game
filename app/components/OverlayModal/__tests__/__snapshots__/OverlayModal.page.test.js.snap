// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverlayModal component renders correctly 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  onModalHide={[Function]}
  onRequestClose={[Function]}
  transparent={true}
  visible={false}
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
    testID={undefined}
  />
  <View
    collapsable={undefined}
    onModalHide={[Function]}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "margin": 37.5,
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#FFF",
          "borderRadius": 30,
          "flex": 0,
        }
      }
    >
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "black",
              "fontSize": 40,
              "margin": 10,
              "paddingVertical": 20,
              "textAlign": "center",
            }
          }
        >
          Time's Up
        </Text>
      </View>
      <View
        style={
          Object {
            "margin": 20,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "flex": 1,
                  "margin": 10,
                },
                Object {
                  "textAlign": "right",
                },
              ]
            }
          >
            Time: 
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "flex": 1,
                  "margin": 10,
                },
                Object {
                  "textAlign": "left",
                },
              ]
            }
          >
            60
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "flex": 1,
                  "margin": 10,
                },
                Object {
                  "textAlign": "right",
                },
              ]
            }
          >
            Score: 
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "flex": 1,
                  "margin": 10,
                },
                Object {
                  "textAlign": "left",
                },
              ]
            }
          >
            0
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "flex": 1,
                  "margin": 10,
                },
                Object {
                  "textAlign": "right",
                },
              ]
            }
          >
            High Score: 
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "flex": 1,
                  "margin": 10,
                },
                Object {
                  "textAlign": "left",
                },
              ]
            }
          >
            12
          </Text>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 30,
              "margin": 20,
              "textAlign": "center",
            }
          }
        >
          Close
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;
